### PR TITLE
A4G prebid adapter

### PR DESCRIPTION
**Pull request by adilets**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ x] New bidder adapter https://github.com/prebid/prebid.github.io/pull/333/files
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

Adding new bidder adapter for A4G

- test parameters for validating bids
```
{
  bidder: 'a4g',
  params: {
    zoneId: 59304,
    deliveryUrl: 'http://dev01.ad4game.com/v1/bid'
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer: adilet@ad4game.com, jaafar@ad4game.com, tech@ad4game.com
- [ ] official adapter submission

### CODE REVIEW

**Comment by matthewlane** [comment]
This adapter should be updated to be compliant with the upcoming 1.0 conventions. See http://prebid.org/dev-docs/bidder-adapter-1.html for information about this. Thanks for the PR

**Comment by adilets** [comment]
@matthewlane could you please review my code. Where should I change ? Thanks

**Comment by adilets** [comment]
@matthewlane, @jaiminpanchal27 I have updated my PR. Could you please review it ?

File: modules/a4gBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      }
+    };
+  },
+
+  interpretResponse: function(serverResponses, request) {
+    const bidResponses = [];
+    utils._each(serverResponses, function(response) {
+      const bidResponse = {
+        requestId: request.bidId,
+        bidderCode: spec.code,
```
Comment (matthewlane) [inline modules/a4gBidAdapter.js]: `bidderCode` can be dropped from the bid response object, it'll be set automatically by `bidderFactory`. Docs just updated, sorry for the confusion

File: modules/a4gBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        [SIZE_PARAM_NAME]: sizeParams.join(ARRAY_PARAM_SEPARATOR),
+        [ID_PARAM_NAME]: idParams.join(ARRAY_PARAM_SEPARATOR),
+        [ZONE_ID_PARAM_NAME]: zoneIds.join(ARRAY_PARAM_SEPARATOR)
+      }
+    };
+  },
+
+  interpretResponse: function(serverResponses, request) {
+    const bidResponses = [];
+    utils._each(serverResponses, function(response) {
```
Comment (matthewlane) [inline modules/a4gBidAdapter.js]: #1748 changed the first argument of `interpretResponse` to:

```JavaScript
{
  body: responseBody,
  headers: {
    get: function(header) { /* returns a header from the HTTP response */ }
  }
}
```

so updating `serverResponses` to `serverResponses.body` (and in corresponding tests if needed) should do the trick. May need to rebase off of master first

**Comment by adilets** [comment]
Hello matthewlane. I have updated my PR. I checked my server and resolved your comment messages.

**Comment by adilets** [comment]
@matthewlane I rebased to master. Maybe i should create new PR

**Comment by matthewlane** [comment]
Yeah, feel free to create a new PR. I think the changes look ok, but somehow it's showing that 105 files were changed. If you're able to figure out what's going on there you can push again to this PR but if it's easier, creating a new one with just your adapter is fine too and I'll finish the review there.

Also, still getting the same CORS error as above when testing the adapter. What I'm doing is taking the a4g params from the .md file and putting them on the [hello_world test page](https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/hello_world.html#L23), running `gulp serve`, and opening the page in a browser. The CORS error appears in the dev console and checking `pbjs.getBidResponses()` in the there shows a `statusMessage` of "Bid returned empty or error response" for `a4g`.  Are you getting different results in this setup?

**Pull request closed without merging**